src/api/auth.js

```diff
@@ -0,0 +1,46 @@
+'use-strict';
+import * as jwk from 'jsonwebtoken';
+import * as jwkToPem from 'jwk-to-pem';
+import * as request from 'request';
+
+const iss = process.env.ISS;
+
+// Reusable Authorizer function, set on `authorizer` field in serverless.yml
+export const authorize = (event, context, cb) => {
+    console.log('Auth function invoked');
+    if (event.authorizationToken) {
+        // Remove 'bearer ' from token:
+        const token = event.authorizationToken.substring(7);
+        // Make a request to the iss + .well-known/jwks.json URL:
+        request(
+            {url: `${iss}.well-known/jwks.json`, json: true},
+            (error, response, body) => {
+                if (error || response.statusCode !== 200) {
+                    console.log('Request error:', error);
+                    cb('Unauthorized');
```

Comment:
Since we are using API proxy integration we need to change the callback function as below 
```javascript
const done = (err, res) => callback(null, {
     statusCode: err ? '400' : '200',
     body: err ? err.message : JSON.stringify(res),
     headers: {
          'Content-Type': 'application/json',
          'Access-Control-Allow-Origin': '*',
     },
   });`